### ISSUE TITLE
Failed to remove vm

### ISSUE THREAD

**Issue by iarovyi** [Azure/terraform-azurerm-compute#121]
- Terraform Version: 12
- OS Version: Windows

Bug description: Module contains provider definition so it fails to remove instance if module is commented out. Described in https://github.com/hashicorp/terraform/issues/22907
Potential fix is to remove provider block so provider will be in root module of code that deploys this module

Steps to reproduce:

1. Deploy simple windows instance
```
module "windowsservers" {
     source              = "Azure/compute/azurerm"
     location            = var.location
     vm_hostname         = local.vm_hostname
     admin_password      = "ComplxP@ssw0rd!"
     vm_os_simple        = "WindowsServer"
     is_windows_image    = "true"
     public_ip_dns       = ["xxx"]
     vnet_subnet_id      = module.network.vnet_subnets[0]
 }
```
2.  Comment out module and apply

Result: It fails with following message
```
module.xxxx.module.windowsservers.azurerm_virtual_machine.vm-windows[0],
after which you can remove the provider configuration again.


Error: Provider configuration not present

To work with
module.xxxx.module.windowsservers.azurerm_public_ip.vm[0] its
original provider configuration at
module.xxxx.module.windowsservers.provider.azurerm is required, but
it has been removed. This occurs when a provider configuration is removed
while objects created by that provider still exist in the state. Re-add the
provider configuration to destroy
module.app-shaping-dev.module.windowsservers.azurerm_public_ip.vm[0], after
which you can remove the provider configuration again.
```

**Comment by yupwei68** [comment]
Hi @iarovyi , Thanks for your suggestion. We'll fix that in the next release

**Comment by yupwei68** [comment]
Hi @iarovyi , we have removed the provider definition. Please check if it's solved in the recent release v3.0.0 while you're availble.
